File: elixir/003/003.exs
defmodule Euler do
  def prime_factors(n) do
    prime_factors(n, 2)
  end

  defp prime_factors(n, p) when n < (p * p) do
    [n]
  end

  defp prime_factors(n, p) when rem(n, p) == 0 do
    [p | prime_factors(div(n, p), p)]
  end

  defp prime_factors(n, p) do
    prime_factors(n, p + 1)
  end
end

IO.puts Euler.prime_factors(600851475143)
|> :lists.max
|> to_string

File: fsharp/003/003.fs
let prime_factors n =
  let rec pf current possible_factor =
    if current < (possible_factor * possible_factor) then [current]
    elif current % possible_factor = 0I then possible_factor :: (pf (current / possible_factor) possible_factor)
    else pf current (possible_factor + 1I)
  pf n 2I

printf "%A\n" ((prime_factors 600851475143I) |> List.max)
